platform/ios/src/MGLMapView.mm

```diff
@@ -3622,61 +3629,71 @@ - (MGLAnnotationTag)annotationTagAtPoint:(CGPoint)point persistingResults:(BOOL)
     queryRect = CGRectInset(queryRect, -MGLAnnotationImagePaddingForHitTest,
                             -MGLAnnotationImagePaddingForHitTest);
     std::vector<MGLAnnotationTag> nearbyAnnotations = [self annotationTagsInRect:queryRect];
+    BOOL queryingShapeAnnotations = NO;
+    
+    if (!nearbyAnnotations.size()) {
+        nearbyAnnotations = [self shapeAnnotationTagsInRect:queryRect];
+        queryingShapeAnnotations = YES;
+    }
 
     if (nearbyAnnotations.size())
     {
         // Assume that the user is fat-fingering an annotation.
         CGRect hitRect = CGRectInset({ point, CGSizeZero },
                                      -MGLAnnotationImagePaddingForHitTest,
                                      -MGLAnnotationImagePaddingForHitTest);
+        
+        if (!queryingShapeAnnotations) {
+            // Filter out any annotation whose image or view is unselectable or for which
+            // hit testing fails.
+            auto end = std::remove_if(nearbyAnnotations.begin(), nearbyAnnotations.end(),
+                                      [&](const MGLAnnotationTag annotationTag)
+                                      {
+                                          id <MGLAnnotation> annotation = [self annotationWithTag:annotationTag];
```

Comment:
The indentation is a bit extreme here. I realize Xcode did this automatically, but would you mind indenting this lambda by only one tab beyond the `auto` instead of aligning with the opening parenthesis of the `remove_if()` call? (In this situation, I’d probably use the ⇧⌥⌘V shortcut to preserve indentation then indent manually.) Alternatively, consider breaking the lambda out into a separate local variable, which will make this code easier to read.